fecgov/fec-cms
# Update boolean search box and add tooltips for AO,MUR, Statutes

## Summary
Make design/layout edits to keyword modal html and jinja templates
Add tooltip for keyword text-box field on AO, MUR, Statues
 - AO Search is React.js, so I made the tooltip in here, a reusable react component (Could now be 
    used on any filter type inside AO search with custom tooltip message and positioning classes)
 - MUR and Statutes datatables are both  rendered from legal.jinja, so the tooltip is hardcoded there

- Resolves #2283

## Impacted areas of the application
AO Search react app
	- modified:   fec/static/js/legal/Filters.js
	- modified:   fec/static/js/legal/filters/TextFilter.js
	- added:       fec/static/js/legal/filters/TooltipHelp.js
MUR and Statutes datatables:
	- modified:   legal/templates/macros/legal.jinja
        - modified:   fec/static/scss/components/_tooltips.scss
Keyword Modal popup:
        - modified:   home/templates/partials/legal-keyword-modal.html
	- modified:   legal/templates/partials/legal-keyword-modal.jinja
        - modified:   fec/static/scss/components/_search-controls.scss
        - modified:   fec/static/scss/components/_modals.scss

## Screenshots
### Tested in Internet Explorer below, per checklist item 5 in issue:
![gif2](https://user-images.githubusercontent.com/5572856/49615414-80ae4b80-f97b-11e8-96f6-06392125dda4.gif)
<hr>


## How to test
Checkout this branch locally: feature/update-boolean-search-box
NPM run build
Runserver and test the tooltips on Keyword text filter on AO, MUR and Statutes searches
Test keyword modal box on the same pages as well as the legal landing page (this uses a different template)
If one were ambitious they could also test the reusable tooltip react component by adding it to `DateFilter.js`* by:
- Copying the tooltip code from the `TextFilter.js` file:
  - Lline 3 
  - Lines 21-27 
  - Line 69 and line 79
...and pasting it into `DateFilter.js`.
- Then enable it in` Filters.js`  for that filter-type (DateFilter), like I did for the  TextFilter (@line 56). Maybe add a different message.
- Don't forget to  run build,  and see if it works! ( I tested it but would still like to see if someone else thinks it works well)

*For filters that depend on other filters, like CheckboxFiler, the process is different, but it can still be added to any filter in the panel.
